torch/overrides.py

```diff
@@ -1384,6 +1385,15 @@ def handle_torch_function(
     # overloaded_args already have unique types.
     types = tuple(map(type, overloaded_args))
 
+    # Check for __torch_function__ mode.
+    mode = _get_torch_function_mode()
+    if mode is not None:
+        # NB: unlike on tensors, modes are instances
+        with _no_torch_function_mode():
```

Comment:
It's all TLS, how could there be a race? (Do you want an assert?)

Reply:
Race was a bad word, it was more that these APIs needs to be done in a specific order. But since they are private and not exposed to the user that's fine.

I think I was also thrown off by the "stack only in python" idea.
Sounds good